config/v1/types_cluster_operator.go

```diff
@@ -0,0 +1,106 @@
+// the rest of the cluster.
+// +genclient
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterOperator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec hold the intent of how this operator should behave.
+	Spec ClusterOperatorSpec `json:"spec"`
+
+	// status holds the information about the state of an operator.  It is consistent with status information across
+	// the kube ecosystem.
+	Status ClusterOperatorStatus `json:"status"`
+}
+
+// ClusterOperatorSpec is empty for now, but you could imagine holding information like "pause".
+type ClusterOperatorSpec struct {
+}
+
+// ClusterOperatorStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ClusterOperatorStatus struct {
+	// conditions describes the state of the operator's reconciliation functionality.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []ClusterOperatorStatusCondition `json:"conditions"`
+
+	// version indicates which version of the operator updated the current
+	// status object.
+	Version string `json:"version"`
+
+	// extension contains any additional status information specific to the
+	// operator which owns this status object.
+	Extension runtime.RawExtension `json:"extension,omitempty"`
+}
+
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
+// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// ClusterOperatorStatusCondition represents the state of the operator's
+// reconciliation functionality.
+// +k8s:deepcopy-gen=true
+type ClusterOperatorStatusCondition struct {
+	// type specifies the state of the operator's reconciliation functionality.
+	Type ClusterStatusConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the time of the last update to the current status object.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the reason for the condition's last transition.  Reasons are CamelCase
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	Message string `json:"message,omitempty"`
+}
+
+// ClusterStatusConditionType is the state of the operator's reconciliation functionality.
+type ClusterStatusConditionType string
+
+const (
+	// OperatorAvailable indicates that the binary maintained by the operator (eg: openshift-apiserver for the
```

Comment:
@derekwaynecarr suggests "operand" for "binary maintained by the operator"

Reply:
Ugh, I hate that word.